SystemServices/SystemServices.cpp

```diff
@@ -263,6 +263,8 @@ void stringToIarmMode(std::string mode, IARM_Bus_Daemon_SysMode_t& iarmMode)
 
 bool setPowerState(std::string powerState)
 {
+    LOGWARN("setPowerState to %s\n", powerState.c_str());
```

Comment:
This PR is raised as part of patch upstream activity.  So, we are not supposed to change any code. Your comment will be addressed in this ticket (https://jira.rdkcentral.com/jira/browse/BCM-1096 ). We also observed the behaviour of standby and lightsleep is different in curl commands and direct setpowerstate commands. We will address all in the same ticket.

Reply:
so bring all the changes together to comcast repo

Reply:
Hi.. @apatel859  Please check this ticket https://jira.rdkcentral.com/jira/browse/SGMM393-181. Please check the comments from  the ticket 
```
Update 2023-10-18: Here are the expectations of the power state bahaviours, based on the discussion with Comcast/Component owner team:
1. Standby should be treated exactly the same as LIGHT_SLEEP

2. In the LIGHT_SLEEP/STANDBY, box is not put into 'real' standby. Instead it should run normally with disabled:

disabled AV pipelines
disabled HDMI output. Other outputs should not be disabled
Logs should be saved to RAM and updated every 30 minutes and saved into the flash
disabled Front panel 
JSON-RPC call to the Thunder should be possible
For the BCM platform it should be the S0 state.

3. DEEP_SLEEP should invoke real deep_sleep mode. I believe we should aim for the S3 or S2 state.

4. DEEP_SLEEP should have possibility to wake the box using WOL packets.

5. Using WOL for the STANDBY/LIGHTSLEEP is not clarified yet

6. Waking up the box from DEEP_SLEEP should put the box into the LIGHT_SLEEP/STANDBY. Then it's up to the 'application' to take further steps I.e wake the box up to the ON state. App action should be queried using getWakeupReason and getLastWakeupKeyCode to determine that.

7. It is unclear yet what power state should be returned in case of STANDBY/LIGHT_SLEEP. It's unknown if should return STANDBY or LIGHT_SLEEP when there is no preferredStandbyMode set and what is the expectation when preferredStandbyMode is set to LIGHT_SLEEP


Pin

 Collapse comment: [Jian Wang](https://jira.rdkcentral.com/jira/secure/ViewProfile.jspa?name=jwang02) added a comment - [19/Oct/23 4:42 PM](https://jira.rdkcentral.com/jira/browse/SGMM393-181?focusedId=377453&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-377453)

Jian Wang added a comment - 19/Oct/23 4:42 PM
STANDBY mapping to Broadcom S0 with AV port disabled.

https://code.rdkcentral.com/r/plugins/gitiles/collaboration/soc/broadcom/yocto_oe/layers/meta-rdk-broadcom-next/+/6760454b98588868bf68d0b32dd2e239edd4f151


Pin

 Collapse comment: Artur Gebicz added a comment - 20/Oct/23 1:53 PM

[Artur Gebicz](https://jira.rdkcentral.com/jira/secure/ViewProfile.jspa?name=a.gebicz) added a comment - [20/Oct/23 1:53 PM](https://jira.rdkcentral.com/jira/browse/SGMM393-181?focusedId=377729&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-377729)
```
Hi,

 

Here is the CR implementing requirements for the LIGHT_SLEEP to behave as the STANDBY mode and adds proper reporting for the onSystemPowerChanged notification.

https://code.rdkcentral.com/r/c/collaboration/soc/broadcom/yocto_oe/layers/meta-rdk-broadcom-next/+/93545

With following CR, the LIGHT_SLEEP and state reporting is fixed. The test is passing this step. However due to https://jira.rdkcentral.com/jira/browse/SGMM393-229 bringing back state to ON gives success: false in the response. However the transition to the ON state seems to be completed.

Reply:
I will arrange a meeting to discuss this.